core/src/main/java/org/fao/geonet/kernel/security/openidconnect/SimpleOidcUser.java

```diff
@@ -0,0 +1,228 @@
+        }
+
+        if (username != null && username.length() > 0) {
+            surname = idToken.getFamilyName();
+            firstname = idToken.getGivenName();
+            email = idToken.getEmail();
+
+
+            if (idToken.getClaims() != null && idToken.getClaims().containsKey(oidcConfiguration.organizationProperty)) {
+                organisation = (String) idToken.getClaims().get(oidcConfiguration.organizationProperty);
+            }
+
+
+            if (idToken.getAddress() != null) {
+                address = new Address();
+                address.setAddress(idToken.getAddress().getStreetAddress());
+                address.setCity(idToken.getAddress().getLocality());
+                address.setState(idToken.getAddress().getRegion());
+                address.setZip(idToken.getAddress().getPostalCode());
+                address.setCountry(idToken.getAddress().getCountry());
+            }
+
+
+            Map<Profile, List<String>> profileGroups = oidcRoleProcessor.getProfileGroups(idToken);
+            if (profileGroups != null && profileGroups.size() > 0) {
+                profile = oidcRoleProcessor.getMaxProfile(profileGroups).name();
+            }
+        }
+    }
+
+    SimpleOidcUser(OIDCConfiguration oidcConfiguration, OIDCRoleProcessor oidcRoleProcessor, Map attributes) {
+        username = (String) attributes.get(oidcConfiguration.getUserNameAttribute());
+        if (username == null)
+            username = (String) attributes.get(StandardClaimNames.PREFERRED_USERNAME);
+        if (username == null) {
+            username = (String) attributes.get(StandardClaimNames.NAME);
+        }
+        if (username != null) {
+            username = org.apache.commons.lang.StringUtils.left(username, 256); //first max 256 chars
+        }
+
+        if (username != null && username.length() > 0) {
+            surname = (String) attributes.get(StandardClaimNames.FAMILY_NAME);
+            firstname = (String) attributes.get(StandardClaimNames.GIVEN_NAME);
+            ;
+            email = (String) attributes.get(StandardClaimNames.EMAIL);
+            ;
+
+
+            if (attributes.containsKey(oidcConfiguration.organizationProperty)) {
+                organisation = (String) attributes.get(oidcConfiguration.organizationProperty);
+            }
+
+
+//            if (attributes.get(StandardClaimNames.ADDRESS) != null) {
+//                Map _address =  (Map) attributes.get(StandardClaimNames.ADDRESS);
+//                address = new Address();
+//                address.setAddress(_address.getStreetAddress());
+//                address.setCity(idToken.getAddress().getLocality());
+//                address.setState(idToken.getAddress().getRegion());
+//                address.setZip(idToken.getAddress().getPostalCode());
+//                address.setCountry(idToken.getAddress().getCountry());
+//            }
```

Comment:
To confirm, is it commented as this information is not available in `attributes`?

Reply:
yes, this didn't work.  Removed commented code.